tools/SourceKit/lib/SwiftLang/SwiftEditor.cpp

```diff
@@ -1749,8 +1749,11 @@ void SwiftEditorDocument::parse(ImmutableTextSnapshotRef Snapshot,
     Impl.SemanticInfo->getInvocation()->applyTo(CompInv);
     Impl.SemanticInfo->getInvocation()->raw(Args, PrimaryFile);
   } else {
+    // Use stdin as a .swift input to satisfy the driver. Note that we don't
+    // use Impl.FilePath here because it may be invalid filename for driver
+    // like "" or "-foobar".
     SmallVector<const char *, 1> Args;
-    Args.push_back(Impl.FilePath.c_str()); // Input
+    Args.push_back("-");
```

Comment:
This can change whether we treat the file as a main.swift file or not; does that affect non-semantic diagnostics/parsing?

Reply:
If I understand correctly, `ParserUnit`(via `SwiftDocumentSyntaxInfo`) always treat the file as a `main.swift`, i.e. [initializes `SourceFile` with `SourceFileKind::Main` mode](https://github.com/apple/swift/blob/c468ca1e48bc850c94adb07d25adbbb046aad6fd/lib/Parse/Parser.cpp#L1009).
I don't think this change affects parsing.